shm/src/Player.cpp

```diff
@@ -0,0 +1,33 @@
+#include "../inc/Player.hpp"
+
+Player::Player(std::unique_ptr<Ship> ship, size_t money, size_t availableSpace)
+    : ship_(std::move(ship)), money_(money), availableSpace_(availableSpace) 
+{}
+
+size_t Player::getSpeed() const {
+    return ship_->getSpeed();
+};
+
+std::shared_ptr<Cargo> Player::getCargo(size_t index) const {
+    return ship_->getCargo(index);
+}
+ 
+/*size_t Player::countAvailableSpace() const {
+    size_t availableSpace = ship_->getCapacity();
+    if (ship_) {
+        for (size_t i = 0; i < ship_->getHowManyCargos(); i++) {
+            availableSpace -= ship_->getCargo(i)->getAmount();
+        }
+    }
+    return availableSpace;
+}*/
+
+size_t Player::countAvailableSpace() const {
+    size_t usedSpace = std::accumulate(ship_->getVectorCargo().cbegin(), ship_->getVectorCargo().cend(), 0);
```

Comment:
Iteration through a vector of shared pointers but int type needed to add up (amount of cargo).
Thus error: invalid operands to binary expression ('int' and 'const std::__1::shared_ptr<Cargo>')
Hint: each vector element contains a cargo which needs to be accessed.

std::accumulate returns int type but size_t used (which is good) - make accumulate to return size_t